extensions/spatial/src/org/sbml/jsbml/validator/offline/constraints/DomainConstraints.java

```diff
@@ -0,0 +1,249 @@
+		  				
+		  				if(!dom.isSetId()) {
+		  					return false;
+		  				}		  				
+		  				if(!dom.isSetDomainType()) {
+		  					return false;
+		  				}
+		  				
+		  				return super.check(ctx, dom);
+		  			}
+		  		};
+		  		break;
+		  	}
+		  	
+		  	case SPATIAL_20804:
+		  	{
+		  		// A Domain object may contain one and only one instance of the ListOfInteriorPoints element.
+		  		// No other elements from the SBML Level 3 Spatial Processes namespaces are permitted on a 
+		  		// Domain object.
+		  		
+		  		func = new ValidationFunction<Domain>() {
+		  			@Override
+		  	        public boolean check(ValidationContext ctx, Domain dom) {
+		  				
+		  				boolean onlyOneList = new DuplicatedElementValidationFunction<Domain>(SpatialConstants.listOfInteriorPoints).check(ctx, dom);
+		  				boolean noOtherElements = new UnknownPackageElementValidationFunction<Domain>(SpatialConstants.shortLabel).check(ctx, dom);
+		  				return (onlyOneList && noOtherElements);
+		  			}
+		  		};
+		  		
+		  		break;
+		  	}
+		  	
+		  	case SPATIAL_20805:
+		  	{
+		  		// The value of the attribute spatial:domainType of a Domain object must be the identifier 
+		  		// of an existing DomainType object defined in the enclosing Model object. 
+		  		
+		  		func = new ValidationFunction<Domain>() {
+		  			
+		  			@Override
+		  			public boolean check(ValidationContext ctx, Domain dom) {
+		  				
+		  				if(dom.isSetDomainType()) {		  					
+		  					Model m = dom.getModel();
+		  					SpatialModelPlugin smp = (SpatialModelPlugin) m.getPlugin(SpatialConstants.shortLabel);		  					
+		  					if(smp.isSetGeometry()) {		  						
+		  						Geometry g = smp.getGeometry();
+		  						DomainType dt = g.getDomainType(dom.getDomainType());
+		  						if(dt == null) {
+				  					return false;
+				  				}
+		  					}			  				
+		  				}
+		  				
+		  				return true;
+		  			}
+		  		};
+		  		
+		  		break;
+		  	}
+		  	
+		  	case SPATIAL_20806:
+		  	{
+		  		// The attribute spatial:name on a Domain must have a value of data type string.
+		  		
+		  		func = new ValidationFunction<Domain>() {
+		  			
+		  			@Override
+		  			public boolean check(ValidationContext ctx, Domain dom) {
+		  				
+		  				// nothing to check as java read any kind of String
+		  		        
+		  		        return true;
+		  			}
+		  		};
+		  	}
+		  	
+		  	case SPATIAL_20807:
+		  	{
+		  		// Apart from the general notes and annotations subobjects permitted on all SBML objects,
+		  		// a ListOfInteriorPoints container object may only contain InteriorPoint objects.
+		  		
+		  		func = new ValidationFunction<Domain>() {
+
+		  		    @Override
+		  		    public boolean check(ValidationContext ctx, Domain dom) {
+		  		    	
+		  		        if (dom.isSetListOfInteriorPoints()) {		  		        	
+		  		            boolean coreElements = new UnknownCoreElementValidationFunction<ListOf<InteriorPoint>>().check(ctx, dom.getListOfInteriorPoints());
```

Comment:
for both 20807 and 20808, you should be able to use the generic Unknown methods to have only one method call instead of two.